llama-index-core/llama_index/core/llms/llm.py

```diff
@@ -485,38 +540,108 @@ async def astream(
 
         return stream_tokens
 
-    def _extend_prompt(
+    # -- Tool Calling --
+
+    def _get_tools_str(self, tools: List[Any]) -> str:
+        tools_str += "\n----\n".join(
+            [tool.metadata.simple_fn_schema_str for tool in tools]
+        )
+
+    def _call_tool(self, tool_selection: ToolSelection, tool: Any) -> Any:
```

Comment:
where is this being used? doesn't have same interface as _call_tool in subclasses

Reply:
Leftover from an old approach, let's remove